frontend/ui/elements/screen_rotation_menu_table.lua

```diff
@@ -61,90 +79,10 @@ When unchecked, the default rotation of the file browser and the default/saved r
-                end,
-                radio = true,
-                callback = function(touchmenu_instance)
-                    UIManager:broadcastEvent(Event:new("SetRotationMode", Screen.DEVICE_ROTATED_UPRIGHT))
-                    if touchmenu_instance then touchmenu_instance:closeMenu() end
-                end,
-                hold_callback = function(touchmenu_instance)
-                    G_reader_settings:saveSetting("fm_rotation_mode", Screen.DEVICE_ROTATED_UPRIGHT)
-                    if touchmenu_instance then touchmenu_instance:updateItems() end
-                end,
-            })
-            table.insert(rotation_table, {
-                text_func = function()
-                    local text = C_("Rotation", "⤸ 90°")
-                    if G_reader_settings:readSetting("fm_rotation_mode") == Screen.DEVICE_ROTATED_CLOCKWISE then
-                        text = text .. "   ★"
-                    end
-                    return text
-                end,
-                checked_func = function()
-                    return Screen:getRotationMode() == Screen.DEVICE_ROTATED_CLOCKWISE
-                end,
-                radio = true,
-                callback = function(touchmenu_instance)
-                    UIManager:broadcastEvent(Event:new("SetRotationMode", Screen.DEVICE_ROTATED_CLOCKWISE))
-                    if touchmenu_instance then touchmenu_instance:closeMenu() end
-                end,
-                hold_callback = function(touchmenu_instance)
-                    G_reader_settings:saveSetting("fm_rotation_mode", Screen.DEVICE_ROTATED_CLOCKWISE)
-                    if touchmenu_instance then touchmenu_instance:updateItems() end
-                end,
-            })
-            table.insert(rotation_table, {
-                text_func = function()
-                    local text = C_("Rotation", "↓ 180°")
-                    if G_reader_settings:readSetting("fm_rotation_mode") == Screen.DEVICE_ROTATED_UPSIDE_DOWN then
-                        text = text .. "   ★"
-                    end
-                    return text
-                end,
-                checked_func = function()
-                    return Screen:getRotationMode() == Screen.DEVICE_ROTATED_UPSIDE_DOWN
-                end,
-                radio = true,
-                callback = function(touchmenu_instance)
-                    UIManager:broadcastEvent(Event:new("SetRotationMode", Screen.DEVICE_ROTATED_UPSIDE_DOWN))
-                    if touchmenu_instance then touchmenu_instance:closeMenu() end
-                end,
-                hold_callback = function(touchmenu_instance)
-                    G_reader_settings:saveSetting("fm_rotation_mode", Screen.DEVICE_ROTATED_UPSIDE_DOWN)
-                    if touchmenu_instance then touchmenu_instance:updateItems() end
-                end,
-            })
+            table.insert(rotation_table, genMenuItem(C_("Rotation", "⤹ 90°"), Screen.DEVICE_ROTATED_COUNTER_CLOCKWISE))
```

Comment:
I thought counter-clockwise was missing but it happened to land on the next line due to being longer. xD